From b0478cce0377eb2ba833f8bef50c4468d08226b2 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mkbestas@gmail.com>
Date: Wed, 19 Jan 2022 23:24:39 +0200
Subject: [PATCH 13/21] Launcher3: Fix search bar UI in All Apps

Commit Ifa9fabe3a8236513fad8030c0bd8ed4d27ebd549 improved the look
of the search bar, however it had a few issues and the overall
look did not match Settings or Widget search style.

The following improvements have been made:
 - Centered the search bar text vertically inside the search bar
 - Fixed poor contrast of the search bar text against the background
 - Added back search icon that is always shown, matching Widget search
 - Changed paddings to match Widget search

Change-Id: Ia127da05adff45f8860f0870817aaa9fde8e171c
---
 res/drawable/all_apps_search_hint.xml    | 20 --------------------
 res/layout/search_container_all_apps.xml | 14 +++++++++-----
 2 files changed, 9 insertions(+), 25 deletions(-)
 delete mode 100644 res/drawable/all_apps_search_hint.xml

diff --git a/res/drawable/all_apps_search_hint.xml b/res/drawable/all_apps_search_hint.xml
deleted file mode 100644
index a066e98d9d..0000000000
--- a/res/drawable/all_apps_search_hint.xml
+++ /dev/null
@@ -1,20 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<!--
-     Copyright (C) 2016 The Android Open Source Project
-
-     Licensed under the Apache License, Version 2.0 (the "License");
-     you may not use this file except in compliance with the License.
-     You may obtain a copy of the License at
-
-          http://www.apache.org/licenses/LICENSE-2.0
-
-     Unless required by applicable law or agreed to in writing, software
-     distributed under the License is distributed on an "AS IS" BASIS,
-     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-     See the License for the specific language governing permissions and
-     limitations under the License.
--->
-<selector xmlns:android="http://schemas.android.com/apk/res/android">
-    <item android:color="@android:color/transparent" android:state_focused="true" />
-    <item android:color="?android:attr/textColorSecondary" android:alpha="?android:attr/disabledAlpha" />
-</selector>
\ No newline at end of file
diff --git a/res/layout/search_container_all_apps.xml b/res/layout/search_container_all_apps.xml
index 0ad13d538b..31183de2ed 100644
--- a/res/layout/search_container_all_apps.xml
+++ b/res/layout/search_container_all_apps.xml
@@ -17,21 +17,25 @@
     xmlns:android="http://schemas.android.com/apk/res/android"
     android:id="@id/search_container_all_apps"
     android:layout_width="match_parent"
-    android:layout_height="@dimen/all_apps_search_bar_field_height"
+    android:layout_height="wrap_content"
     android:layout_centerHorizontal="true"
     android:layout_gravity="top|center_horizontal"
     android:background="@drawable/bg_all_apps_searchbox"
     android:focusableInTouchMode="true"
+    android:gravity="center_vertical"
     android:hint="@string/all_apps_search_bar_hint"
     android:imeOptions="actionSearch|flagNoExtractUi"
     android:importantForAutofill="no"
     android:inputType="text|textNoSuggestions|textCapWords"
     android:maxLines="1"
-    android:padding="8dp"
-    android:paddingStart="16dp"
+    android:paddingVertical="12dp"
+    android:paddingStart="12dp"
+    android:paddingEnd="0dp"
+    android:drawablePadding="8dp"
+    android:drawableStart="@drawable/ic_allapps_search"
     android:saveEnabled="false"
     android:scrollHorizontally="true"
     android:singleLine="true"
     android:textColor="?android:attr/textColorSecondary"
-    android:textColorHint="@drawable/all_apps_search_hint"
-    android:textSize="20sp" />
+    android:textColorHint="?android:attr/textColorSecondary"
+    android:textSize="20sp" />
\ No newline at end of file
-- 
2.34.1

